extension/parquet/parquet_reader.cpp

```diff
@@ -343,16 +345,33 @@ unique_ptr<ColumnReader> ParquetReader::CreateReader(const duckdb_parquet::forma
 	D_ASSERT(next_schema_idx == file_meta_data->schema.size() - 1);
 	D_ASSERT(file_meta_data->row_groups.empty() || next_file_idx == file_meta_data->row_groups[0].columns.size());
 
+	auto &root_struct_reader = (StructColumnReader &)*ret;
+
 	// add casts if required
 	for (auto &entry : cast_map) {
 		auto column_idx = entry.first;
 		auto &expected_type = entry.second;
-
-		auto &root_struct_reader = (StructColumnReader &)*ret;
 		auto child_reader = move(root_struct_reader.child_readers[column_idx]);
 		auto cast_reader = make_unique<CastColumnReader>(move(child_reader), expected_type);
 		root_struct_reader.child_readers[column_idx] = move(cast_reader);
 	}
+
+	if (parquet_options.filename) {
+		Value val = Value(file_name);
+		root_struct_reader.child_readers.push_back(make_unique<GeneratedConstantColumnReader>(
+		    *this, LogicalType::VARCHAR, SchemaElement(), next_file_idx, 0, 0, val));
+	}
+
+	if (parquet_options.hive_partitioning) {
+		auto res = ParseHivePartitions(file_name);
+
+		for (auto &partition : res) {
```

Comment:
Can we add a test for a mismatch in the number of hive partitions, e.g.:

```sql
SELECT * FROM read_parquet([
   'data/normal_folder/test.parquet',
   'data/a=1/test.parquet'
], HIVE_PARTITIONING=1);
```

Or:

```sql
SELECT * FROM read_csv([
   'data/*/test.csv'
], HIVE_PARTITIONING=1);
```

(where we have two files `data/normal_folder/test.csv` and `data/a=1/test.csv`)